### PR TITLE
Ins 2973 logicrunner sender

### PR DESCRIPTION
**Pull request by sexyhamster**
**- What I did**
moved messages to watermill

**- How I did it**
with my bare brain

**- How to verify it**
run tests

**- Description for the changelog**

### CODE REVIEW

File: logicrunner/executionbroker.go [hunk 1]
```diff
@@ -45,7 +46,7 @@ type ExecutionBrokerI interface {
 	PendingState() insolar.PendingState
 	PrevExecutorStillExecuting(ctx context.Context)
 	PrevExecutorPendingResult(ctx context.Context, prevExecState insolar.PendingState)
-	PrevExecutorFinishedPending(ctx context.Context) error
+	PrevExecutorPendingFinished(ctx context.Context) error
```
Comment (ruz) [inline logicrunner/executionbroker.go]: previous name was valid english, new name is wrong :) Either use old or PrevExecutorSentPendingIsFinished.

File: logicrunner/executionregistry/execution_registry.go [hunk 2]
```diff
@@ -107,23 +107,25 @@ func (r *executionRegistry) Done(transcript *common.Transcript) bool {
...
 		for requestRef := range r.registry {
 			requestRefs = append(requestRefs, requestRef)
 		}
 
-		messages = append(messages, &message.StillExecuting{
-			Reference:   r.objectRef,
-			Executor:    r.jetCoordinator.Me(),
-			RequestRefs: requestRefs,
-		})
+		if len(requestRefs) > 0 {
```
Comment (ruz) [inline logicrunner/executionregistry/execution_registry.go]: this is bogus check. it can not be empty.

Comment (sexyhamster) [reply logicrunner/executionregistry/execution_registry.go]: it can, at least in tests

Comment (ruz) [reply logicrunner/executionregistry/execution_registry.go]: it's impossible. you have `if len(registry) != 0` a few lines above.

File: logicrunner/handle_calls.go [hunk 3]
```diff
@@ -73,7 +73,7 @@ func (h *HandleCall) sendToNextExecutor(
 		ObjectReference: objectRef,
 		RequestRef:      requestRef,
 		Request:         request,
-		ServiceData:     serviceDataFromContext(ctx),
+		ServiceData:     oldDerviceDataFromContext(ctx),
```
Comment (ruz) [inline logicrunner/handle_calls.go]: Dervice?

File: logicrunner/logicrunner.go [hunk 4]
```diff
@@ -281,41 +281,50 @@ func (lr *LogicRunner) stopIfNeeded(ctx context.Context) {
...
 	spanMessages.AddAttributes(trace.StringAttribute("numMessages", strconv.Itoa(len(messages))))
 
 	var sendWg sync.WaitGroup
-	sendWg.Add(len(messages))
 
-	for _, msg := range messages {
-		go lr.sendOnPulseMessage(ctx, msg, &sendWg)
+	for ref, msg := range messages {
+		for _, msg := range msg {
+			sendWg.Add(1)
```
Comment (ruz) [inline logicrunner/logicrunner.go]: `sendWg.Add(len(msg))`

Comment (sexyhamster) [reply logicrunner/logicrunner.go]: fixed

File: logicrunner/executionbroker.go (hunk 1 above)
Comment (bigbes) [reply logicrunner/executionbroker.go]: `PrevExecutorPendingIsFinished` is fine too.

**Pull request merged**